core/org.wso2.carbon.tomcat.ext/src/main/java/org/wso2/carbon/tomcat/ext/valves/RequestCorrelationIdValve.java

```diff
@@ -104,11 +106,16 @@ public void invoke(Request request, Response response) throws IOException, Servl
             }
 
             if (associateToThreadMap.size() == 0) {
-                associateToThread(UUID.randomUUID().toString());
```

Comment:
Basically with the change of this line, there is no behaviour change. It's the same thing you have done which was done in associateToThread private method. Either you can keep this as it was or remove the private method associateToThread which is no longer used :)

Reply:
Also we can element two if blocks with same condition  if (associateToThreadMap.size() == 0) {

Reply:
+1

Reply:
Even there are two blocks Also we can element two if blocks with same condition if (associateToThreadMap.size() == 0) {
they evaluate to different result at different times.
First one tries to read correlation ID from query params, if headers does not contain one. 
Second one is the fall back when it does not find correlation ID from previous step.
That logic is intentional.

Reply:
+1